system-helper/flatpak-system-helper.c

```diff
@@ -1151,6 +1302,355 @@ handle_run_triggers (FlatpakSystemHelper   *object,
+                {
+                  *out_repo_location = g_strdup (repo_path);
+                  return TRUE;
+                }
+            }
+        }
+    }
+
+   return FALSE;
+}
+
+static gboolean
+check_partial_pull_availability (const gchar *repo_tmp,
+                                 const gchar *ref_basename,
+                                 const gchar *commit,
+                                 gchar      **out_location)
+{
+  g_autoptr(GFileEnumerator) enumerator = NULL;
+  g_autoptr(GFile) repo_tmpfile = NULL;
+  GFileInfo *file_info = NULL;
+  g_autofree gchar *prefix = NULL;
+  g_autoptr(GError) error = NULL;
+  const gchar *name;
+
+  g_debug ("Checking partial pull availability for ref %s (commit: %s)", ref_basename, commit);
+
+  repo_tmpfile = g_file_new_for_path (repo_tmp);
+  enumerator = g_file_enumerate_children (repo_tmpfile,
+                                          G_FILE_ATTRIBUTE_STANDARD_NAME,
+                                          G_FILE_QUERY_INFO_NONE, NULL, NULL);
+  if (enumerator == NULL)
+    {
+      g_warning ("Failed to get file enumerator: %s", error->message);
+      return FALSE;
+    }
+
+   prefix = g_strdup_printf ("flatpak-cache-%s", ref_basename);
+
+   /* We are looking for flatpak-cache-$ref-XXXXXX directory in repo/tmp.
+      If we find one, we try to look for an OSTree repo which possibly has
+      the $commit.commitpartial file denoting the partial pull for that $commit. */
+  while (TRUE)
+    {
+      if (!g_file_enumerator_iterate (enumerator, &file_info, NULL, NULL, &error))
+        {
+          g_warning ("Error while file enumerator iterating: %s", error->message);
+          break;
+        }
+
+      if (file_info == NULL)
+        break;
+
+      name = g_file_info_get_name (file_info);
+      if (g_file_info_get_file_type (file_info) == G_FILE_TYPE_DIRECTORY &&
+          g_str_has_prefix (name, prefix))
+        {
+          g_autoptr(GFile) cache_dir = g_file_get_child (repo_tmpfile, name);
+          OngoingPull *pull;
+          gboolean dir_in_use = FALSE;
+
+          /* If this cache directory is used by any other ongoing pull, we
+           * should not be re(using) it. */
+          G_LOCK (ongoing_pulls);
+          for (guint i = 0; i < ongoing_pulls->len; i++)
+            {
+              pull = g_ptr_array_index (ongoing_pulls, i);
+              if (g_strcmp0 (pull->src_dir, name) == 0)
+                {
+                  dir_in_use = TRUE;
+                  break;
+                }
+            }
+          G_UNLOCK (ongoing_pulls);
+
+           if (!dir_in_use && check_for_repo_containing_partial_pull (cache_dir, commit, out_location))
+            return TRUE;
+        }
+    }
+
+  return FALSE;
+}
+
+static gboolean
+handle_get_revokefs_fd (FlatpakSystemHelper   *object,
+                        GDBusMethodInvocation *invocation,
+                        GUnixFDList           *arg_fdlist,
+                        guint                  arg_flags,
```

Comment:
Are these flags actually used? I guess `FLATPAK_HELPER_GET_REVOKEFS_FD_FLAGS_NO_INTERACTION` is for future-proofing and consistency. Nonetheless, this function needs a check that `(arg_flags & ~FLATPAK_HELPER_GET_REVOKEFS_FD_FLAGS_ALL) == 0` like the other methods.